TACC/Core-Portal
# Task/fp 515 install cortal icons

## Overview: ##

1. __New__: Install new custom icons.
1. __Fix__: Update old custom icons to bew new custom icons.
1. __Fix__: Update relevant (i.e. failing) tests.
1. _Minor_: Separate icon font styles from icon styles.

## PR Status: ##


## Related Jira tickets: ##

* [FP-123](https://jira.tacc.utexas.edu/browse/FP-123)

## Summary of Changes: ##

- _Minor_: Disable linting of `icon.fonts.css`.
- __Fix__: Update several files (and tests) to use new icon names:
    - `AppIcon`
    - `AppBrowser`
    - `DataFilesListingCell`
    - `DataFilesSidebar`
    - `DataFilesToolbar`
    - `Sidebar`
    - `TicketModal`
- __New__: Replace old Core Portal font with Cortal Icons font.
- __New__: Migrate font styles to their own stylesheet.
- __Fix__: Add icon aliases:
    - `icon-collapse` (`icon-contract`)
- __Fix__: Support WOFF in Webpack config.

## Testing Steps: ##

1. In Dashboard, check these icons:
    - Sidebar "Dashboard"
    - Sidebar "Data Files"
    - Sidebar "Applications"
    - Sidebar "Allocations"
1. In (Dashboard) My Tickets, check these icons:
    - Add Ticket Modal "Upload Files"
    - View Ticket Modal ticket expand and ticket collapse
1. In Data Files, check these icons:
    - "+ Add" dropdown "Folder"
    - "+ Add" dropdown "Upload"
    - Upload Modal
    - Toolbar "Rename"
    - Toolbar "Move"
    - Toolbar "Copy"
    - Toolbar "Download"
    - Toolbar "Trash"
    - Table folder row
1. In Applications, check these icons:
    - Browser category
    - Browser app (default icon)
    - Browser app (custom icon) i.e. Jupyter

## UI Photos:

<img width="927" alt="Dashboard Sidebar" src="https://user-images.githubusercontent.com/62723358/87366641-cf9c6980-c53e-11ea-9087-1f8a83f0852a.png">
<img width="927" alt="Ticket Add" src="https://user-images.githubusercontent.com/62723358/87366672-dd51ef00-c53e-11ea-9e42-1f51a0b37031.png">
<img width="927" alt="Ticket View" src="https://user-images.githubusercontent.com/62723358/87366680-e216a300-c53e-11ea-93e3-9a3d1e248e03.png">
<img width="927" alt="Data Files Left" src="https://user-images.githubusercontent.com/62723358/87366694-ea6ede00-c53e-11ea-9df9-26b058ee9b41.png">
<img width="927" alt="Data Files Upload" src="https://user-images.githubusercontent.com/62723358/87366744-0a060680-c53f-11ea-80f7-372bea5fb667.png">
<img width="927" alt="Data Files Right" src="https://user-images.githubusercontent.com/62723358/87366699-ec38a180-c53e-11ea-8505-2e914f77f431.png">
<img width="927" alt="Applications Default" src="https://user-images.githubusercontent.com/62723358/87366711-f35faf80-c53e-11ea-8a18-91077951929b.png">
<img width="927" alt="Applications Custom" src="https://user-images.githubusercontent.com/62723358/87366722-f6f33680-c53e-11ea-94a8-b35fb44e9473.png">

## Notes: ##
